src/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/InitialDataTci.hpp

```diff
@@ -0,0 +1,72 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/Tags/Inactive.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOptions.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+template <size_t Dim>
+class Mesh;
+template <typename TagsList>
+class Variables;
+/// \endcond
+
+namespace grmhd::GhValenciaDivClean::subcell {
+/*!
+ * \brief The troubled-cell indicator run on DG initial data to see if we need
+ * to switch to subcell.
+ *
+ * The following checks are done in the order they are listed:
+ *
+ * - if `TildeD` (`TildeTau`) on the DG or subcell grid (projected from the DG
+ *   grid, not initialized to the initial data) is less than
```

Comment:
If either the DG or the FD data is bad we switch. This is because if you set up a discontinuity on the DG grid it will have positive density, but the projected/interpolated density might be negative because the polynomial is very oscillatory. Does that make sense?